src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -179,6 +180,11 @@ public void RegisterWorkspaceHost(IVisualStudioWorkspaceHost host)
             _workspaceHosts.Add(new WorkspaceHostState(this, host));
         }
 
+        public Task RegisterWorkspaceHostFromForegroundThreadAsync(IVisualStudioWorkspaceHost host)
```

Comment:
@jasonmalinowski my question probably is that, that thread affinity is imposed by project tracker itself rather than by others, so why not itself make sure that happens when it needs and remove that thread affinity requirement for callers?